### PR TITLE
restore: implement DeltaRestoreOperations methods

### PR DESCRIPTION
**Pull request by derekbit**
https://github.com/longhorn/longhorn/issues/6576

Based on https://github.com/longhorn/backupstore/pull/145

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -69,3 +69,5 @@ require (
 	k8s.io/mount-utils v0.27.1 // indirect
 	k8s.io/utils v0.0.0-20230406110748-d93618cff8a2 // indirect
 )
+
+replace github.com/longhorn/backupstore v0.0.0-20230830075002-fa25b1a97ffd => github.com/derekbit/backupstore v0.0.0-20230906134210-074ab4a06389
```
Comment (FrankYang0529) [inline go.mod:73]: The https://github.com/longhorn/backupstore/pull/145 is merged. I think we can just update `longhorn/backupstore` version?

**Review by james-munson — APPROVED** [review]
The changes look fine - mostly moving code, but adding the implementation for DeltaBlockBackupOperations interface.  It'd be good to get this into main, and fix the current breakage with top-of-main backupstore.

**Comment by derekbit** [comment]
cc @shuo-wu @PhanLe1010 @innobead

**Comment by PhanLe1010** [comment]
@derekbit There is still a reference to your private repo. Could you fix it? Thank you:
```
replace github.com/longhorn/backupstore v0.0.0-20230830075002-fa25b1a97ffd => github.com/derekbit/backupstore v0.0.0-20230906134210-074ab4a06389
```

**Comment by derekbit** [comment]
> @derekbit There is still a reference to your private repo. Could you fix it? Thank you:
> 
> ```
> replace github.com/longhorn/backupstore v0.0.0-20230830075002-fa25b1a97ffd => github.com/derekbit/backupstore v0.0.0-20230906134210-074ab4a06389
> ```

Updated. Thank you

**Review by PhanLe1010 — APPROVED** [review]

**Review by shuo-wu — APPROVED** [review]

**Pull request merged**